src/main/java/com/pusher/client/util/Factory.java

```diff
@@ -119,6 +121,13 @@ public synchronized void shutdownThreads() {
         }
     }
 
+    public synchronized Gson getGson() {
+        if (gson == null) {
```

Comment:
There's very little advantage to a lazy instantiation here, and significant potential cost in synchronizing the method.

Given that the methods we're using on the Gson object are pure, I'd suggest simply putting it in a constant, which also simplifies the dependency injection dance that's been required for this change set.

In order to keep the coupling down, rather than a public constant in `Factory`, why not just create a `Gson` object in a private constant in each class which requires one?